src/main/java/com/iexec/worker/metric/ComputeDurationsService.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2023-2023 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.worker.metric;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import org.apache.commons.math.stat.descriptive.DescriptiveStatistics;
+
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Simple service to store durations of a compute stage (pre, app or post-compute)
+ * and aggregates them (min, max and average).
+ */
+public class ComputeDurationsService {
+    private static final int STATISTICS_WINDOW = 100;
+
+    private final Map<String, Long> durationPerChainTaskId = new LinkedHashMap<>();
+    private final DescriptiveStatistics statistics;
+
+    public ComputeDurationsService(MeterRegistry registry,
+                                   String workerWalletAddress,
+                                   String context) {
+        statistics = new DescriptiveStatistics(STATISTICS_WINDOW);
```

Comment:
I think this could be done out of the constructor as well, directly at instantiation.
If you prefer to keep it in constructor, it is fine as well.

Reply:
Done in [c0f6e3e](https://github.com/iExecBlockchainComputing/iexec-worker/pull/559/commits/c0f6e3e146e75f10252091940b25eed9680a7eed).
But it finally seemed more logic to me to let the user parameterize the window. So I got back to the initialization in the constructor, with a new parameter for the window. Please see [bebfef2](https://github.com/iExecBlockchainComputing/iexec-worker/pull/559/commits/bebfef2b6479760c00ff3484df3284d3e4da395e).